profile/profile.go

```diff
@@ -109,10 +109,11 @@ type Mapping struct {
 
 // Location corresponds to Profile.Location
 type Location struct {
-	ID      uint64
-	Mapping *Mapping
-	Address uint64
-	Line    []Line
+	ID       uint64
+	Mapping  *Mapping
+	Address  uint64
+	Line     []Line
+	IsFolded bool
```

Comment:
One thing to check is that this flag should be reset when Line array is reset in the symbolization scenarios.

Reply:
I think I took care of this in internal/symbolizer/symbolizer.go, but let me know if my understanding here is wrong.